debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -267,7 +271,9 @@ private void stop(boolean restart) {
     public void commitRecord(SourceRecord record) throws InterruptedException {
         Map<String, ?> currentOffset = record.sourceOffset();
         if (currentOffset != null) {
-            this.lastOffset = currentOffset;
+            stateLock.lock();
+            lastOffsets.put(record.sourcePartition(), currentOffset);
+            stateLock.unlock();
```

Comment:
It may be a good idea to extract these three lines into a separate `private` method.

Reply:
✔️